lib/core/util.js

```diff
@@ -495,6 +495,24 @@ function isValidHTTPToken (characters) {
   return true
 }
 
+// headerCharRegex have been lifted from
+// https://github.com/nodejs/node/blob/main/lib/_http_common.js
+
+/**
+ * Matches if val contains an invalid field-vchar
+ *  field-value    = *( field-content / obs-fold )
+ *  field-content  = field-vchar [ 1*( SP / HTAB ) field-vchar ]
+ *  field-vchar    = VCHAR / obs-text
+ */
+const headerCharRegex = /[^\t\x20-\x7e\x80-\xff]/
+
+/**
+ * @param {string} characters
+ */
+function isValidHeaderChar (characters) {
+  return !headerCharRegex.test(characters)
+}
```

Comment:
interesting.

i wonder how the performance is if we flip it and do it like this

const isInvalidHeaderChar = headerCharRegex.test.bind(headerCharRegex)

Reply:
There is no change.
```

benchmark                     time (avg)             (min … max)       p75       p99      p999
---------------------------------------------------------------- -----------------------------
• isValidHeaderChar# text/html
---------------------------------------------------------------- -----------------------------
regexp.test                33.69 ns/iter     (28.27 ns … 297 ns)  31.45 ns  79.35 ns    143 ns
regexp.exec                35.91 ns/iter     (31.64 ns … 273 ns)  34.62 ns  80.37 ns    153 ns
charCodeAt                 27.58 ns/iter        (25 ns … 532 ns)  25.88 ns  62.16 ns    133 ns
isValidHeaderChar          32.33 ns/iter      (29.1 ns … 452 ns)  30.62 ns  78.03 ns    163 ns
isInvalidHeaderCharRegex   30.87 ns/iter     (28.22 ns … 195 ns)  30.03 ns  66.36 ns    133 ns

summary for isValidHeaderChar# text/html
  charCodeAt
   1.12x faster than isInvalidHeaderCharRegex
   1.17x faster than isValidHeaderChar
   1.22x faster than regexp.test
   1.3x faster than regexp.exec

• isValidHeaderChar# application/json; charset=UTF-8
---------------------------------------------------------------- -----------------------------
regexp.test                59.13 ns/iter      (54.2 ns … 443 ns)   56.4 ns    120 ns    229 ns
regexp.exec                64.15 ns/iter     (58.35 ns … 268 ns)  60.64 ns    134 ns    212 ns
charCodeAt                 97.63 ns/iter     (89.45 ns … 355 ns)  98.97 ns    180 ns    303 ns
isValidHeaderChar           74.7 ns/iter     (57.71 ns … 357 ns)     75 ns    170 ns    277 ns
isInvalidHeaderCharRegex   59.08 ns/iter     (54.25 ns … 280 ns)  56.25 ns    147 ns    245 ns

summary for isValidHeaderChar# application/json; charset=UTF-8
  isInvalidHeaderCharRegex
   1x faster than regexp.test
   1.09x faster than regexp.exec
   1.26x faster than isValidHeaderChar
   1.65x faster than charCodeAt
```

Reply:
thank you for the investigation